internal/telemetry/client.go

```diff
@@ -225,14 +236,46 @@ func (c *Client) Stop() {
 		return
 	}
 	c.started = false
-	c.flushT.Stop()
 	c.heartbeatT.Stop()
 	// close request types have no body
 	r := c.newRequest(RequestTypeAppClosing)
 	c.scheduleSubmit(r)
 	c.flush()
 }
 
+// ProductEnabled enqueues an app-product-change event that signals a product has been turned on/off.
+// the caller can also specify additional configuration changes (e.g. profiler config info),
+// which will be sent via the app-client-configuration-change event
+func (c *Client) ProductEnabled(namespace Namespace, enabled bool, configuration []Configuration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if !c.started {
+		c.log("attempted to send product change event, but telemetry client has not started")
+		return
+	}
+	productReq := c.newRequest(RequestTypeAppProductChange)
+	products := new(Products)
+	switch namespace {
+	case NamespaceProfilers:
+		products.Profiler = ProductDetails{Enabled: enabled}
+	case NamespaceASM:
+		products.AppSec = ProductDetails{Enabled: enabled}
+	default:
+		c.log("unknown")
```

Comment:
let's make this more descriptive

Reply:
I remember writing a complete message here, might've accidentally deleted it my bad!